JetNews/app/src/main/java/com/example/jetnews/ui/home/PostCardYourNetwork.kt

```diff
@@ -33,53 +34,91 @@ import androidx.ui.material.ripple.Ripple
 import androidx.ui.res.imageResource
 import androidx.ui.text.style.TextOverflow
+import androidx.ui.tooling.preview.Preview
 import androidx.ui.unit.dp
 import com.example.jetnews.R
+import com.example.jetnews.data.post1
 import com.example.jetnews.model.Post
+import com.example.jetnews.model.PostAuthor
 import com.example.jetnews.ui.Screen
 import com.example.jetnews.ui.navigateTo
 
+private val cardSize = LayoutSize(280.dp, 240.dp)
+
 @Composable
-fun PostCardPopular(post: Post) {
-    Card(shape = RoundedCornerShape(4.dp)) {
+fun PostCardPopular(post: Post, modifier: Modifier = Modifier.None) {
+    Card(shape = RoundedCornerShape(4.dp), modifier = modifier) {
         Ripple(bounded = true) {
             Clickable(onClick = {
                 navigateTo(Screen.Article(post.id))
             }) {
-                Container(modifier = LayoutSize(280.dp, 240.dp)) {
-                    Column(modifier = LayoutSize.Fill) {
-                        val image = post.image ?: imageResource(R.drawable.placeholder_4_3)
-                        Container(modifier = LayoutHeight(100.dp) + LayoutSize.Fill) {
-                            DrawImage(image)
-                        }
-                        Column(modifier = LayoutPadding(16.dp)) {
-
-                            val emphasisLevels = EmphasisLevels()
-                            ProvideEmphasis(emphasis = emphasisLevels.high) {
-                                Text(
-                                    text = post.title,
-                                    style = MaterialTheme.typography().h6,
-                                    maxLines = 2,
-                                    overflow = TextOverflow.Ellipsis
-                                )
-                            }
-                            ProvideEmphasis(emphasis = emphasisLevels.high) {
-                                Text(
-                                    text = post.metadata.author.name,
-                                    style = MaterialTheme.typography().body2
-                                )
-                            }
-                            ProvideEmphasis(emphasis = emphasisLevels.high) {
-                                Text(
-                                    text = "${post.metadata.date} - " +
-                                            "${post.metadata.readTimeMinutes} min read",
-                                    style = MaterialTheme.typography().body2
-                                )
-                            }
-                        }
+                Column(modifier = cardSize) {
+                    val image = post.image ?: imageResource(R.drawable.placeholder_4_3)
+                    Container(modifier = LayoutHeight(100.dp) + LayoutSize.Fill) {
+                        DrawImage(image)
                     }
+                    BottomSection(post)
                 }
             }
         }
     }
 }
+
+@Composable
+private fun BottomSection(post: Post) {
```

Comment:
While I like factoring this out, we're drifting further from the [tutorial](https://developer.android.com/jetpack/compose/tutorial). Keep as is?

Reply:
+1 will revert